sentry-kotlin-multiplatform-gradle-plugin/src/main/java/io/sentry/kotlin/multiplatform/gradle/FrameworkPathResolver.kt

```diff
@@ -0,0 +1,256 @@
+            }
+        }
+
+        return searchDirectory(startingDir)
+    }
+}
+
+/**
+ * Performs a manual search for Sentry Cocoa frameworks using system tools.
+ *
+ * This strategy:
+ * - Searches the DerivedData for valid framework paths
+ * - Returns first validated paths found for either static or dynamic frameworks
+ *
+ * If multiple paths were found for a single framework type, the most recently used is chosen.
+ * See [ManualFrameworkPathSearchValueSource] for details.
+ */
+class ManualSearchStrategy(
+    private val project: Project,
+    private val basePathToSearch: String? = null
+) : FrameworkResolutionStrategy {
+    // TODO: currently the search doesnt differentiate between Cocoa versions
+    // we can improve this by checking the info.plist and prefer the ones that are the version we are looking for
+    override fun resolvePaths(architectures: Set<String>): FrameworkPaths {
+        val dynamicValueSource =
+            project.providers.of(ManualFrameworkPathSearchValueSource::class.java) {
+                it.parameters.frameworkType.set(FrameworkType.DYNAMIC)
+                if (basePathToSearch != null) {
+                    it.parameters.basePathToSearch.set(basePathToSearch)
+                }
+            }
+        val staticValueSource =
+            project.providers.of(ManualFrameworkPathSearchValueSource::class.java) {
+                it.parameters.frameworkType.set(FrameworkType.STATIC)
+                if (basePathToSearch != null) {
+                    it.parameters.basePathToSearch.set(basePathToSearch)
+                }
+            }
+
+        return FrameworkPaths.createValidated(
+            dynamicBasePath = dynamicValueSource.orNull,
+            staticBasePath = staticValueSource.orNull,
+            architectures = architectures
+        )
+    }
+}
+
+class FrameworkPathResolver(
+    private val project: Project,
+    private val strategies: List<FrameworkResolutionStrategy> = defaultStrategies(project)
+) {
+    fun resolvePaths(
+        architectures: Set<String>
+    ): FrameworkPaths {
+        strategies.forEach { strategy ->
+            try {
+                project.logger.info(
+                    "Attempt to resolve Sentry Cocoa framework paths using ${strategy::class.simpleName}"
+                )
+                val result = strategy.resolvePaths(architectures)
+                if (result != FrameworkPaths.NONE) {
+                    val path = result.dynamic ?: result.static
+                    project.logger.lifecycle("Found Sentry Cocoa framework path using ${strategy::class.simpleName} at $path")
+                    return result
+                } else {
+                    project.logger.debug("Strategy ${strategy::class.simpleName} did not find valid paths")
+                }
+            } catch (e: FrameworkLinkingException) {
+                project.logger.warn(
+                    "Strategy ${strategy::class.simpleName} failed due to error: ${e.message}"
```

Comment:
there's a `warn` overload that accepts `Throwable` as a parameter - should we use it here to pass the exception to have it print the stacktrace nicely?